### ISSUE TITLE
systemd-gpt-auto-generator overrides swap options in fstab

### ISSUE THREAD

**Issue by evelineraine** [systemd/systemd#6192]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> 233

### Used distribution

> Fedora 26 Beta

### In case of bug report: Expected behaviour you didn't see

> Swap enabling with TRIM on boot automatically.

### In case of bug report: Unexpected behaviour you saw

> Swap enabling without TRIM on boot. But _with_ TRIM when activated using `swapon`.

### In case of bug report: Steps to reproduce the problem

> 1. Use GPT
> 2. Set swap partition's type to 'Linux swap'
> 3. Add `discard` option to swap entry in fstab
> 4. Reboot

When swap entry is present in fstab and a swap GPT partition exists, systemd ends up with 2 dynamic swap units in /run: one from systemd-gpt-auto-generator and one from systemd-fstab-generator. The former unit seems to take precedence, so fstab options for swap are effectively ignored. However, using `swapon` yields the correct results, since it apparently uses fstab directly.

Workaround is to change GPT partition type to something other than `Linux swap`. This results in only one unit being generated (by fstab-generator) and swap activates properly on boot.

When using MBR, no such issue occurs.

Unit files:
```
# /run/systemd/generator/dev-disk-by\x2duuid-cdd6f778\x2d00c8\x2d4a13\x2da19a\x2d405a0bbe95d3.swap
# Automatically generated by systemd-fstab-generator

[Unit]
SourcePath=/etc/fstab
Documentation=man:fstab(5) man:systemd-fstab-generator(8)

[Swap]
What=/dev/disk/by-uuid/cdd6f778-00c8-4a13-a19a-405a0bbe95d3
Options=discard

# /run/systemd/generator.late/dev-sda3.swap
# Automatically generated by systemd-gpt-auto-generator

[Unit]
Description=Swap Partition
Documentation=man:systemd-gpt-auto-generator(8)

[Swap]
What=/dev/sda3
```

fstab entry for swap:
```
UUID=cdd6f778-00c8-4a13-a19a-405a0bbe95d3 swap        swap    discard     0 0
```

**Comment by poettering** [comment]
hmm, I figure we should disable swap auto discovery automatically if there's at least one swap partition listed in /etc/fstab.

**Comment by poettering** [comment]
note that the original idea was that user config will override auto-discovered config. But that falls flat if the device node is referenced in a different way, i.e. once via the uuid name in fstab, and the other way via a raw kernel device name the way gpt-auto-discovery does it.

**Comment by poettering** [comment]
Fix waiting in #5976
